--- a/age/of/civilizations2/jakowski/lukasz/Commands.java
+++ b/age/of/civilizations2/jakowski/lukasz/Commands.java
@@ -2,0 +3 @@
+import age.of.modifications.flafmg.mod.ModManager;
@@ -3,0 +5,2 @@
+
+import java.io.*;
@@ -52,0 +56,14 @@
+               if(tempCommand[0].equals("reload")){
+                  addMessage("reloading mods..");
+                  CFG.toast.setInView("Reloading");
+
+                  CFG.modManager = new ModManager();
+
+                  if (CFG.modManager.getModPaths() != null) {
+                     for (File modPath : CFG.modManager.getModPaths()) {
+                        CFG.modManager.loadMod(modPath.getPath());
+                     }
+                  }
+                  return;
+               }
+
@@ -830 +846,0 @@
-
