--- a/age/of/civilizations2/jakowski/lukasz/Menu_MapEditor_WastelandMaps_Edit.java
+++ b/age/of/civilizations2/jakowski/lukasz/Menu_MapEditor_WastelandMaps_Edit.java
@@ -184 +184 @@
-         case 7:
+         case 7: {
@@ -193,2 +193,2 @@
-                  && CFG.game.getProvince(ixxx).getWasteland() < 0
-                  && CFG.game.getProvince(ixxx).getMinX() < tMinX) {
+                       && CFG.game.getProvince(ixxx).getWasteland() < 0
+                       && CFG.game.getProvince(ixxx).getMinX() < tMinX) {
@@ -203 +203,2 @@
-         case 8:
+         }
+         case 8: {
@@ -220 +221,2 @@
-         case 9:
+         }
+         case 9: {
@@ -237 +239,2 @@
-         case 10:
+         }
+         case 10: {
@@ -253,0 +257 @@
+         }
