--- a/age/of/civilizations2/jakowski/lukasz/Menu_InitGame.java
+++ b/age/of/civilizations2/jakowski/lukasz/Menu_InitGame.java
@@ -3,0 +4 @@
+import age.of.modifications.flafmg.mod.*;
@@ -12,0 +14,2 @@
+
+import java.io.*;
@@ -19 +22 @@
-   private int iNumOfSteps = 33;
+   private int iNumOfSteps = 34;
@@ -21,0 +25,2 @@
+   private int loadedMods = 0;
+   private File[] modPaths = ModManager.getModPaths();
@@ -29 +33,0 @@
-
@@ -92 +96,7 @@
-            SteamManager.init();
+            //AOM: added this for debug pouposes ONLY,
+            //this will disable the steam check so i dont need to be with steam open all the time
+            try{
+               SteamManager.init();
+            } catch (Exception e){
+               e.printStackTrace();
+            }
@@ -480,0 +491,21 @@
+            //load mods
+            if(modPaths == null || modPaths.length == 0) {
+                this.iStepID++;
+                return;
+            }
+            CFG.sLoading = "Loading Mods "
+                    + " [#"
+                    + (loadedMods + 1)
+                    + "/"
+                    + modPaths.length
+                    + "]: "
+                    + modPaths[loadedMods].getName()
+                    + " - ";
+            CFG.modManager.loadMod(modPaths[loadedMods].getAbsolutePath());
+            loadedMods++;
+
+            if(loadedMods >= modPaths.length){
+               this.iStepID++;
+            }
+
+         } else if (this.iStepID == 33 + CFG.map.getMapNumOfProvinces(CFG.map.getActiveMapID()) * 2) {
@@ -499 +530,2 @@
-
+            CFG.modManager.loaded = true;
+            CFG.modManager.callOnPostInit();
