--- a/age/of/civilizations2/jakowski/lukasz/Steam_Game.java
+++ b/age/of/civilizations2/jakowski/lukasz/Steam_Game.java
@@ -229,3 +228,0 @@
-               @Override
-               public void onAuthSessionTicket(SteamAuthTicket authTicket, SteamResult result) {
-               }
@@ -357,4 +353,0 @@
-                  @Override
-                  public void onNumberOfCurrentPlayersReceived(boolean success, int players) {
-                  }
-
